### DISCUSSION TITLE
[@types/node] `new URL(input)` should accept `URL`s

### DISCUSSION THREAD

**Discussion by danielbankhead** [DefinitelyTyped/DefinitelyTyped#68454] [Issues with a @types package] [answered]
## Summary

There's a minor type issue where `new URL(input)` does not accept `URL`s:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/eaaf5e44d4c874c4f7e38210cc62d053292f3b13/types/node/url.d.ts#L415

Should be:
```ts
constructor(input: string | URL, base?: string | URL);
```

Technically it should support other stringifiers, but may not be necessary for this fix.

## Documentation

- https://nodejs.org/api/url.html#new-urlinput-base
- https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters

## Examples/Use Cases

The follow example is valid in Node.js, but is not with `@types/node`:
```js
const input = new URL('https://example.com');
const output = new URL(input);
```

This can be helpful in real-world use cases where an API may want to accept a URL as a `string | URL`:
```ts
export interface Config {
  /**
   * An endpoint to use.
   *
   * @example
   * 'https://localhost:8080/'
   */
  endpoint: string | URL;
}

function resolveURL (config: Config): URL {
  // currently a type error today
  return new URL(config.endpoint);
}
```

[a comment by an automated account is not shown]

**Comment by ghost** [comment, marked as answer]
Fixed in PR #68456
